hummingbot/client/config/strategy_config_data_types.py

```diff
@@ -0,0 +1,171 @@
+            raise ValueError(ret)
+        return v
+
+
+class BaseTradingStrategyMakerTakerConfigMap(BaseStrategyConfigMap):
+    maker_market: str = Field(
+        default=...,
+        description="The name of the maker exchange connector.",
+        client_data=ClientFieldData(
+            prompt=lambda mi: "Enter your maker spot connector",
+            prompt_on_new=True,
+        ),
+    )
+    taker_market: str = Field(
+        default=...,
+        description="The name of the taker exchange connector.",
+        client_data=ClientFieldData(
+            prompt=lambda mi: "Enter your taker spot connector",
+            prompt_on_new=True,
+        ),
+    )
+    maker_market_trading_pair: str = Field(
+        default=...,
+        description="The name of the maker trading pair.",
+        client_data=ClientFieldData(
+            prompt=lambda mi: BaseTradingStrategyMakerTakerConfigMap.trading_pair_prompt(mi, True),
+            prompt_on_new=True,
+        ),
+    )
+    taker_market_trading_pair: str = Field(
+        default=...,
+        description="The name of the taker trading pair.",
+        client_data=ClientFieldData(
+            prompt=lambda mi: BaseTradingStrategyMakerTakerConfigMap.trading_pair_prompt(mi, False),
+            prompt_on_new=True,
+        ),
+    )
+
+    @classmethod
+    def trading_pair_prompt(cls, model_instance: 'BaseTradingStrategyMakerTakerConfigMap', is_maker: bool) -> str:
+        if is_maker:
+            exchange = model_instance.maker_market
+            example = AllConnectorSettings.get_example_pairs().get(exchange)
+            market_type = "maker"
+        else:
+            exchange = model_instance.taker_market
+            example = AllConnectorSettings.get_example_pairs().get(exchange)
+            market_type = "taker"
+        return (
+            f"Enter the token trading pair you would like to trade on {market_type} market:"
+            f" {exchange}{f' (e.g. {example})' if example else ''}"
+        )
+
+    @validator(
+        "maker_market",
+        "taker_market",
+        pre=True
+    )
+    def validate_exchange(cls, v: str, field: Field):
+        """Used for client-friendly error output."""
+        ret = validate_exchange(v)
+        if ret is not None:
+            raise ValueError(ret)
+        if field.name == "maker_market_trading_pair":
+            cls.__fields__["maker_market"].type_ = ClientConfigEnum(  # rebuild the exchanges enum
+                value="Exchanges",  # noqa: F821
+                names={e: e for e in AllConnectorSettings.get_connector_settings().keys()},
+                type=str,
+            )
+        if field.name == "taker_market_trading_pair":
+            cls.__fields__["taker_market"].type_ = ClientConfigEnum(  # rebuild the exchanges enum
+                value="Exchanges",  # noqa: F821
+                names={e: e for e in AllConnectorSettings.get_connector_settings().keys()},
+                type=str,
+            )
```

Comment:
These should also be removed with the new approach

Reply:
I removed it for both fields